crates/wasmparser/src/validator/core/canonical.rs

```diff
@@ -246,6 +246,29 @@ pub(crate) trait InternRecGroup {
                     }
                 }
             }
+            CompositeInnerType::Cont(t) => {
+                if !features.stack_switching() {
+                    bail!(
+                        offset,
+                        "cannot define continuation types when stack switching is disabled",
+                    );
+                }
+                if !features.gc_types() {
+                    bail!(
+                        offset,
+                        "cannot define continuation types when gc types are disabled",
+                    );
+                }
+                // Check that the type index points to a valid function type.
+                let id = match t.0.as_core_type_id() {
+                    None => bail!(offset, "invalid continuation type index"),
+                    Some(id) => id,
+                };
```

Comment:
Here it's ok to `.unwrap()` the result since a failure here means a bug in wasmparser itself rather than something the user can fix with the input wasm module